### ISSUE TITLE
ESlint breaks on `application/json` type script tags having valid JSON

### ISSUE THREAD

**Issue by umakantozy** [eslint/eslint#13757]
**Tell us about your environment**

* **ESLint Version:** v7.10.0 (Currently used)
* **Node Version:** v14.3.0
* **npm Version:**  v6.14.5

**What parser (default, `@babel/eslint-parser`, `@typescript-eslint/parser`, etc.) are you using?**: babel-eslint

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```json
{
  "root": true,
  "env": {
    "browser": true,
    "es6": true,
    "jest": true,
    "node": false
  },
  "extends": ["eslint:recommended", "plugin:react/recommended"],
  "parser": "babel-eslint",
  "plugins": [
    "react",
    "react-hooks"
  ],
  "settings": {
    "react": {
      "version": "16.13.0"
    }
  },
  "rules": {
    "react/no-string-refs": "warn",
    "react/jsx-no-target-blank": "warn",
    "react/no-unescaped-entities": "off",
    "react-hooks/rules-of-hooks": "error",
    "react-hooks/exhaustive-deps": "warn",
    "no-undef": "warn",
    "accessor-pairs": "error",
    "array-bracket-spacing": [
      "error",
      "never"
    ],
    "array-callback-return": "error",
    "arrow-body-style": "error",
    "arrow-parens": [
      "error",
      "as-needed"
    ],
    "arrow-spacing": [
      "error",
      {
        "after": true,
        "before": true
      }
    ],
    "block-scoped-var": "error",
    "block-spacing": "error",
    "brace-style": "error",
    "callback-return": "error",
    "camelcase": ["error", {"allow": ["featured_video", "featured_audio"]}],
    "comma-dangle": "error",
    "comma-spacing": [
      "error",
      {
        "after": true,
        "before": false
      }
    ],
    "comma-style": [
      "error",
      "last"
    ],
    "complexity": "error",
    "computed-property-spacing": [
      "error",
      "never"
    ],
    "consistent-return": "off",
    "consistent-this": "error",
    "curly": "error",
    "default-case": "error",
    "dot-location": [
      "error",
      "property"
    ],
    "dot-notation": "error",
    "eol-last": "off",
    "eqeqeq": "error",
    "func-names": [
      "error",
      "never"
    ],
    "func-style": "off",
    "generator-star-spacing": "error",
    "global-require": "error",
    "guard-for-in": "error",
    "handle-callback-err": "error",
    "id-blacklist": "error",
    "id-length": "off",
    "id-match": "error",
    "indent": ["error", 2, {"SwitchCase": 1}],
    "init-declarations": "off",
    "jsx-quotes": "error",
    "key-spacing": "error",
    "keyword-spacing": "error",
    "linebreak-style": [
      "error",
      "unix"
    ],
    "lines-around-comment": "error",
    "max-depth": "error",
    "max-len": "off",
    "max-lines": ["error", 1400],
    "max-nested-callbacks": "error",
    "max-params": ["warn", 8],
    "max-statements": ["error", 46],
    "max-statements-per-line": "error",
    "multiline-ternary": "off",
    "new-cap": "error",
    "new-parens": "error",
    "newline-after-var": "off",
    "newline-before-return": "off",
    "newline-per-chained-call": "off",
    "no-alert": "error",
    "no-array-constructor": "error",
    "no-bitwise": "error",
    "no-caller": "error",
    "no-catch-shadow": "error",
    "no-confusing-arrow": "off",
    "no-continue": "error",
    "no-div-regex": "error",
    "no-duplicate-imports": "error",
    "no-else-return": "error",
    "no-empty-function": "error",
    "no-eq-null": "error",
    "no-eval": "error",
    "no-extend-native": "error",
    "no-extra-bind": "error",
    "no-extra-label": "error",
    "no-extra-parens": ["off", "all", {
      "conditionalAssign": true,
      "nestedBinaryExpressions": false,
      "returnAssign": false,
      "ignoreJSX": "multi-line",
      "enforceForArrowConditionals": false
    }],
    "no-floating-decimal": "error",
    "no-implicit-coercion": "off",
    "no-implicit-globals": "error",
    "no-implied-eval": "error",
    "no-invalid-this": "off",
    "no-iterator": "error",
    "no-label-var": "error",
    "no-labels": "error",
    "no-lone-blocks": "error",
    "no-lonely-if": "error",
    "no-loop-func": "error",
    "no-magic-numbers": "off",
    "no-mixed-operators": "error",
    "no-mixed-requires": "error",
    "no-mixed-spaces-and-tabs": ["error", "smart-tabs"],
    "no-multi-spaces": "error",
    "no-multi-str": "error",
    "no-multiple-empty-lines": "error",
    "no-negated-condition": "error",
    "no-nested-ternary": "error",
    "no-new": "error",
    "no-new-func": "error",
    "no-new-object": "error",
    "no-new-require": "error",
    "no-new-wrappers": "error",
    "no-octal-escape": "error",
    "no-param-reassign": "error",
    "no-path-concat": "error",
    "no-plusplus": "error",
    "no-process-env": "error",
    "no-process-exit": "error",
    "no-proto": "error",
    "no-prototype-builtins": "error",
    "no-restricted-globals": "error",
    "no-restricted-imports": "error",
    "no-restricted-modules": "error",
    "no-restricted-syntax": "error",
    "no-return-assign": "off",
    "no-script-url": "error",
    "no-self-compare": "error",
    "no-sequences": "error",
    "no-shadow": "error",
    "no-shadow-restricted-names": "error",
    "no-spaced-func": "error",
    "no-sync": "off",
    "no-ternary": "off",
    "no-throw-literal": "error",
    "no-trailing-spaces": [
      "error",
      {
        "skipBlankLines": true
      }
    ],
    "no-undef-init": "error",
    "no-undefined": "error",
    "no-underscore-dangle": ["error", { "allow": ["__REDUX_DEVTOOLS_EXTENSION__"] }],
    "no-unmodified-loop-condition": "error",
    "no-unneeded-ternary": [
      "error",
      {
        "defaultAssignment": true
      }
    ],
    "no-unused-expressions": "off",
    "no-use-before-define": "error",
    "no-useless-call": "error",
    "no-useless-computed-key": "error",
    "no-useless-concat": "error",
    "no-useless-constructor": "error",
    "no-useless-escape": "error",
    "no-useless-rename": "error",
    "no-var": "off",
    "no-void": "error",
    "no-warning-comments": "off",
    "no-whitespace-before-property": "error",
    "no-with": "error",
    "object-curly-newline": "off",
    "object-curly-spacing": [
      "error",
      "never"
    ],
    "object-property-newline": "off",
    "object-shorthand": "error",
    "one-var": "off",
    "one-var-declaration-per-line": "error",
    "operator-assignment": "error",
    "operator-linebreak": "error",
    "padded-blocks": "off",
    "prefer-arrow-callback": "off",
    "prefer-const": "off",
    "prefer-reflect": "off",
    "prefer-rest-params": "error",
    "prefer-spread": "off",
    "prefer-template": "error",
    "quote-props": "off",
    "quotes": [
      "error",
      "single",
      {"allowTemplateLiterals": true}
    ],
    "radix": "error",
    "require-jsdoc": ["off", {
      "require": {
        "FunctionDeclaration": true,
        "MethodDefinition": true,
        "ClassDeclaration": true
      }
    }],
    "rest-spread-spacing": "error",
    "semi": ["error", "always"],
    "semi-spacing": "error",
    "sort-imports": "off",
    "sort-vars": "off",
    "space-before-blocks": "error",
    "space-before-function-paren": "off",
    "space-in-parens": [
      "error",
      "never"
    ],
    "space-infix-ops": "error",
    "space-unary-ops": "error",
    "spaced-comment": [
      "error",
      "always"
    ],
    "strict": "error",
    "template-curly-spacing": "error",
    "unicode-bom": [
      "error",
      "never"
    ],
    "valid-jsdoc": "warn",
    "vars-on-top": "error",
    "wrap-iife": "error",
    "wrap-regex": "error",
    "yield-star-spacing": "error",
    "yoda": "error"
  }
}
```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```html
<amp-analytics type="gtag">
   <script type="application/json">
        {
            "vars": {}
        }
   </script>
</amp-analytics>   
```

```bash
node_modules/.bin/eslint 'app/**/*.{js,jsx}' 'server/**/*.{js,jsx}'
```

**What did you expect to happen?**
Parse, show the result of eslint errors or successful pass.

**What actually happened? Please include the actual, raw output from ESLint.**
/Users/myuser/myproject/app/src/some-component/SomeComponent.jsx
```bash
  265:19  error  Parsing error: Unexpected token, expected "}"

  263 |         <script type="application/json">
  264 |           {
> 265 |             "vars": {}
      |                   ^
  266 |           }
  267 |         </script>
  268 |       </amp-analytics>}

✖ 1 problem (1 error, 0 warnings)
```

**Are you willing to submit a pull request to fix this bug?**

Sorry, NO!

**Comment by ljharb** [comment]
I'm confused; eslint doesn't lint HTML files, which is the only place script tags can occur. You also can't have a script tag inline in jsx like that afaik, without using `__dangerouslySetInnerHTML`.

**Comment by umakantozy** [comment]
Oh yes. That is good catch { } do wrap a JS expression in JSX files. Closing the issue. Thanks for clarification.
